File: ruby/shuffle_string.rb
str = 'codeleet'
indices = [4, 5, 6, 7, 0, 2, 1, 3]

def restore_string(str, indices)
  answer = []
  indices.each_with_index do |n, i|
    answer[n] = str[i]
  end

  answer.join
end

p restore_string(str, indices)

File: python/shuffle_string.py
str = "codeleet"
indices = [4, 5, 6, 7, 0, 2, 1, 3]


def restore_string(str, indices):
  answer = [None] * len(indices)


  for i, val in enumerate(indices):
    answer[val] = str[i]

  return ''.join(answer)

print(restore_string(str, indices))

